### PR TITLE
Disable masking on Aarch64 without SVE support

### PR DESCRIPTION
**Pull request by dcaballe**
Masking is a catch-all transformation that only requires vector sizes to be provided to the vectorizer. This patch makes sure that we don't provide vector sizes to the vectorizer when we don't want to explicitly enable vector masking (e.g., Aarch64 without SVE support, at least for now).

benchmarks: x86_64

### CODE REVIEW

**Review by vmurali — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 1]
```diff
@@ -436,6 +437,7 @@ void addDoubleTilingPadExpertPassPipeline(OpPassManager &passManager) {
   {
     LinalgSingleTilingExpertPassOptions options;
     options.vectorize = true;
+    options.enableVectorMasking = true;
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp]: THis is set to true unconditionally?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
LGTM but maybe there is one small bug here.

**Review by MaheshRavishankar — APPROVED** [review]

**Comment by dcaballe** [comment]
Hey @pzread, it looks like the x86 benchmarks were executed successfully but the report wasn't posted. Android ones seem to be failing due to a cancelation signal after a timeout. I run them both multiple times. Anything going on?

**Review by hanhanW — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp [hunk 2]
```diff
@@ -654,6 +661,7 @@ void addMmt4dTilingExpertPassPipeline(OpPassManager &passManager) {
   {
     LinalgSingleTilingExpertPassOptions options;
     options.vectorize = true;
+    options.vectorize = enableVectorMasking;
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp]: This should be `options.enableVectorMasking = enableVectorMasking;`

**Comment by dcaballe** [comment]
Masking is a catch-all transformation that only requires vector sizes to be provided to the vectorizer. This patch makes sure that we don't provide vector sizes to the vectorizer when we don't want to explicitly enable vector masking (e.g., Aarch64 without SVE support, at least for now).

**Pull request merged**